src/feed.tsx

```diff
@@ -8,18 +9,64 @@ export interface Properties {
   app: string;
 }
 
-export class Feed extends React.Component<Properties> {
+export interface State {
+  feed: FeedItemModel[];
+  hasMore?: boolean;
+  pageNumber?: number;
+  pageSize?: number;
+}
+
+export class Feed extends React.Component<Properties, State> {
+  componentDidMount() {
+    this.setState({ feed: this.props.items.slice(0, 2) });
+  }
+
+  componentDidUpdate(prevProps: Properties) {
+    const { items } = this.props;
+    if (items !== prevProps.items) {
+      this.setState({ feed: items.slice(0, 2) });
+      this.setState({ pageNumber: 2, pageSize: 2, hasMore: true });
```

Comment:
i don't think this will function how we want it to. for the above tests i mentioned, let's make sure that adjusting the items in different ways results in the expected changes (or non-changes) to both the rendered items, as well as the props we're passing to the `InfiniteScroll`